pkg/volume/volume.go

```diff
@@ -65,6 +66,12 @@ type Recycler interface {
 	Recycle() error
 }
 
+// Create adds a new resource in the storage provider and creates a PersistentVolume for the new resource.
+// Calls to Create should block until complete.
+type Creater interface {
+	Create() (*api.PersistentVolume, error)
```

Comment:
I think now this is a little bit backward or should at least be 2 operations:

1.  the provisioner needs to provide a template PV for it's plugin according to the volumeOptions.
2.  the provisioner needs to invoke its internal logic for creation of the resource in the infrastructure.

A Claim's provisioning should be like this:

1. Create PVC with experimental annotation.  will only bind to a PV with a matching pre-bind annotation.
2. Watch claim, Create PV - but unfulfilled in infrastructure.  special annotation prevents binding until fulfillment.
3. Watch volumes, create resource -- use real PV name to link to provider -- remove unfulfilled annotation and save real volumeID.
4. claim will bind to newly provisioned volume